flink-runtime/src/main/java/org/apache/flink/runtime/io/AvailabilityListener.java

```diff
@@ -55,7 +50,7 @@
 	 * @return a future that is completed if there are more records available. If there are more
 	 * records available immediately, {@link #AVAILABLE} should be returned. Previously returned
 	 * not completed futures should become completed once there is more input available or if
-	 * the input {@link #isFinished()}.
+	 * the input is finished.
```

Comment:
Should we also have a constant for `FINISHED`? As is the `AvailabilityListener` depends on its implementations to provide some `isFinished()`.

Reply:
Not quite sure of it.  `isFinished()` is more proper to be provided by input interface to indicate whether this input ends or not. As for `AvailabilityListener`, we just explain that which cases would cause future complete. Or I did not get your key point?